src/lib/LedgerApi.ts

```diff
@@ -284,7 +285,14 @@ class LedgerApi {
         const request = new LedgerApiRequest(LedgerApi.RequestType.CONFIRM_ADDRESS,
             async (api, params): Promise<string> => {
                 const result = await api.getAddress(params.keyPath, /*validate*/ true, /*display*/ true);
-                return result.address;
+                const confirmedAddress = result.address;
+
+                if (params.addressToConfirm!.replace(/ /g, '').toUpperCase()
+                    !== confirmedAddress.replace(/ /g, '').toUpperCase()) {
+                    LedgerApi._throwError(LedgerApi.ErrorType.REQUEST_ASSERTION_FAILED, 'Address mismatch', request);
+                }
```

Comment:
Does `LedgerApi._throwError` actually `throw` an `Error`? Or do you need to `return` here, to avoid resolving this request?

Reply:
Yes, it actually throws.